src/validations/trip.validation.js

```diff
@@ -45,41 +42,29 @@ async function tripValidation(req, res, next) {
       }),
     accommodationId: Joi.number()
   });
-
   const { error } = schema.validate(req.body);
-  const errorMessages = [];
   if (error) {
-    error.details.forEach((err) => {
-      errorMessages.push(err.message.replace(/[^a-zA-Z0-9 .-]/g, ''));
-    });
-  }
-
-  if (req.body.originId && req.body.destinationId) {
-    const locations = [req.body.originId, req.body.destinationId];
-    await Promise.all(locations.map(async loc => {
-      const isLocationExist = await LocationService.findLocationByProperty({ id: loc });
-      if (!isLocationExist) {
-        errorMessages.push(`location with id ${loc} is not available`);
-      }
-    }));
-  }
-
-  if (departureDate > returnDate) {
-    ResponseService.setError(400, 'Travel date can not be greater than return date');
-    return ResponseService.send(res);
-  }
-  if (errorMessages.length !== 0) {
+    const errorMessages = error.details.map((err) => err.message.replace(/[^a-zA-Z0-9 .-]/g, ''));
     ResponseService.setError(400, errorMessages);
     return ResponseService.send(res);
   }
-
-  const tripId = `${req.userData.id}${new Date().getFullYear()}${new Date().getMonth()}${new Date().getDate()}${req.body.destinationId}`;
-  const existTrip = await TripService.findTripByProperty({ tripId });
-  if (existTrip) {
+  const locations = [req.body.originId, req.body.destinationId];
+  await Promise.all(locations.map(async loc => {
+    const isLocationExist = await LocationService.findLocationByProperty({ id: loc });
+    if (!isLocationExist) {
+      ResponseService.setError(400, `location with id ${loc} is not available`);
+      return ResponseService.send(res);
+    }
+  }));
+  const tripExist = await TripService.findTripByProperty({
+    originId: req.body.originId,
+    destinationId: req.body.destinationId,
+    departureDate: { [Op.between]: [new Date(`${req.body.departureDate}T00:00:00.000Z`), new Date(`${req.body.departureDate}T23:59:59.999Z`)] }
+  });
```

Comment:
What's the difference between this and multi-city trip validation?

Reply:
I'm not sure if you are asking me with those lines of code or if you are asking between 2 files of multi-city and that one of trip which was supposed to be return trip validation.  But the difference is that multi city validation hasn't returnDate field but return trip validation does. other difference is that the body for return trip is object and for multi city it is an array. If I didn't answer this accordingly, please rephrase the question.

Reply:
My concern here is that there is a lot of boilerplate codes in trip, one-way and multi-city trips validation files.

Abstract the codes into a function and reuse it in the other files